### PR TITLE
adds variant calling

### PR DESCRIPTION
**Pull request by jemten**
Adds a variantcalling workflow to the pipeline. The user can choose between bcftools (default) and gatk. Bcftools call can either be run in consensus or multi allelic (default) mode.
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/genomic-medicine-sweden/tomte/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/tomte _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jemten** [comment]
Sorry for the long PR @lucpen, classic case of scope creep. No rush with this one, we can merge your PR first

File: CITATIONS.md [hunk 1]
```diff
@@ -10,8 +10,16 @@
 
 ## Pipeline tools
 
+- [BCFtools](https://academic.oup.com/gigascience/article/10/2/giab008/6137722)
+
+  > Danecek P, Bonfield JK, Liddle J, et al. Twelve years of SAMtools and BCFtools. GigaScience. 2021;10(2):giab008. doi:10.1093/gigascience/giab008
+
```
Comment (Lucpen) [inline CITATIONS.md:16]:
```suggestion
- [BCFtools](https://academic.oup.com/gigascience/article/10/2/giab008/6137722)

  > Danecek P, Bonfield JK, Liddle J, et al. Twelve years of SAMtools and BCFtools. GigaScience. 2021;10(2):giab008. doi:10.1093/gigascience/giab008

- [SAMtools](https://academic.oup.com/gigascience/article/10/2/giab008/6137722)

  > Danecek P, Bonfield JK, Liddle J, et al. Twelve years of SAMtools and BCFtools. GigaScience. 2021;10(2):giab008. doi:10.1093/gigascience/giab008
 
```

Comment (Lucpen) [reply CITATIONS.md:16]: Just wanted to add SAMtools too and this was the easiest way, as the paper and link are the same

**Review by Lucpen — APPROVED** [review]
LGTM! I just added a suggestion because it seems an easy way to include it and it needs to be added :)

**Pull request merged**